File: examples/python/single_token/single_token.py
# The 'requests' module is available via pip: "pip install requests"
# You can find more documentation about requests at http://docs.python-requests.org/en/latest/
import requests
import json
import locale
import sys

STORE_DOMAIN = "<< YOUR STORE DOMAIN >>"
ACCESS_TOKEN = "<< YOUR ACCESS TOKEN >>"

locale.setlocale(locale.LC_ALL, '')

def get_product_list(token):
    headers = {
        'X-AC-Auth-Token': ACCESS_TOKEN
    }
    uri = 'https://{}/api/v1/products'.format(STORE_DOMAIN)
    # include verify=False if using dev certificate
    # r = requests.get(uri, headers = headers, verify = False)
    r = requests.get(uri, headers = headers)
    return r.json()


if __name__ == '__main__':
    product_list = get_product_list(ACCESS_TOKEN)

    for product in product_list['products']:
        encoded_name = product['item_name'].encode('utf-8')
        line_end = ": {}\n".format(locale.currency(product['price']), grouping = True).encode('utf-8')
        sys.stdout.buffer.write(encoded_name + line_end)

File: examples/golang/single_token/single_token.go
package main

import (
	"encoding/json"
	"fmt"
	"io/ioutil"
	"net/http"
)

const storeDomain = "<< YOUR STORE DOMAIN >>"
const accessToken = "<< YOUR ACCESS TOKEN >>"

type Product struct {
	ItemName string `json:"item_name"`
	Price    float64
}

type ProductList struct {
	NextPage     string `json:"next_page"`
	PreviousPage string `json:"previous_page"`
	TotalCount   int    `json:"total_count"`
	Products     []Product
}

func getProductList() (*ProductList, error) {
	req, err := http.NewRequest("GET", fmt.Sprintf("https://%s/api/v1/products", storeDomain), nil)
	if err != nil {
		return nil, err
	}
	req.Header.Add("X-AC-Auth-Token", accessToken)

	client := new(http.Client)
	resp, err := client.Do(req)
	if err != nil {
		return nil, err
	}
	defer resp.Body.Close()

	body, err := ioutil.ReadAll(resp.Body)
	if err != nil {
		return nil, err
	}

	var list ProductList
	err = json.Unmarshal(body, &list)
	if err != nil {
		return nil, err
	}
	return &list, err
}

func main() {
	list, err := getProductList()
	if err != nil {
		fmt.Println(err)
	} else {
		for _, p := range list.Products {
			fmt.Println(p.ItemName+":", p.Price)
		}
	}
}
